### ISSUE TITLE
Issue with distributed training for larger batch sizes and for more workers.

### ISSUE THREAD

**Issue by anirudh2290** [eric-haibin-lin/mxnet#180]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System: Linux 

Compiler: gcc 4.8.3

Package used (Python/R/Scala/Julia): Python

MXNet version: sparse branch

## Error Message:

terminate called after throwing an instance of 'dmlc::Error'
  what():  [18:23:51] src/io/./iter_sparse_batchloader.h:85: Check failed: sparse_base_->Next() number of input must be bigger than batch size

Stack trace returned 7 entries:
[bt] (0) /myEFSvolume/sparse_jun/python/mxnet/../../lib/libmxnet.so(_ZN4dmlc15LogMessageFatalD1Ev+0x29) [0x7f95d05024b9]
[bt] (1) /myEFSvolume/sparse_jun/python/mxnet/../../lib/libmxnet.so(_ZN5mxnet2io17SparseBatchLoader4NextEv+0xe6e) [0x7f95d132e54e]
[bt] (2) /myEFSvolume/sparse_jun/python/mxnet/../../lib/libmxnet.so(_ZZN5mxnet2io20SparsePrefetcherIter4InitERKSt6vectorISt4pairISsSsESaIS4_EEENKUlPPNS_9DataBatchEE_clESB_+0x2b) [0x7f95d1331c9b]
[bt] (3) /myEFSvolume/sparse_jun/python/mxnet/../../lib/libmxnet.so(_ZZN4dmlc12ThreadedIterIN5mxnet9DataBatchEE4InitESt8functionIFbPPS2_EES4_IFvvEEENKUlvE_clEv+0x1f4) [0x7f95d12ab0a4]
[bt] (4) /usr/lib64/libstdc++.so.6(+0xb1ba0) [0x7f95c9373ba0]
[bt] (5) /lib64/libpthread.so.0(+0x7dc5) [0x7f95dd08bdc5]
[bt] (6) /lib64/libc.so.6(clone+0x6d) [0x7f95dc6b06ed]

## Minimum reproducible example

Running distributed training with criteo dataset for a simple model with batch_size: 32784, output_dim: 32 and softmax layer as an output on a cluster of 12 workers and one master.
ssh -A -X <USER>@<HOST> "(python /myEFSvolume/mxnet/tools/launch.py -n 12-H \$DEEPLEARNING_WORKERS_PATH python /myEFSvolume/sparse_jun/benchmark/python/sparse_end2end.py --dataset criteo --log-level DEBUG --num-gpu 0 --kvstore dist_sync --num-epoch 3 --batch-size 32784 --output-dim 32)"

https://github.com/anirudh2290/mxnet/blob/dist_bench_changes/benchmark/python/sparse_end2end.py#L78

## What have you tried to solve it?

1. Tried increasing the criteo dataset size to the following. Initially it was lesser  
I am using 12 workers and the batch_size is 32784 so total batches = 32784 * 12 = 393408
The total number of lines in criteo.t.
cat criteo.t | wc -l
548787

**Comment by eric-haibin-lin** [comment]
Closed now since it cannot be reproduced.
